oauth2client/client.py

```diff
@@ -1222,6 +1227,31 @@ def create_scoped(self, scopes):
         """
         return self
 
+    @classmethod
+    def from_json(cls, s):
+        from oauth2client.service_account import _ServiceAccountCredentials
+        s = _from_bytes(s)
+        data = json.loads(s)
+
+        # A GoogleCredentials.from_stream return value might be a
```

Comment:
I think this is the sticking point for me - I'm a strong adherent to the "a superclass' having any knowledge of any of its subclasses is an abstraction violation" design principle. Are you also an adherent? If so, do you agree that there's a problem here? If there's a problem here, are we making it much worse with this change or only a little worse? If there's a problem here, is there a fix we can manage now in this change or should we put it off until later?